### ISSUE TITLE
The examples should be updated to support desktop

### ISSUE THREAD

**Issue by GroovinChip** [flutter/flutter#84306]
None of the example applications found in `/examples` support desktop. 

I am working on this in my fork, so feel free to assign this issue to me.

**Comment by TahaTesser** [comment]
Hi @GroovinChip 
That's true, however, this could break a bunch of tests

Some examples:
https://github.com/flutter/flutter/blob/5cc7b6c3ea5fd76111920fec8803712e06fda98c/dev/devicelab/bin/tasks/devtools_profile_start_test.dart
https://github.com/flutter/flutter/blob/34059eec2c671472ee95f5093705fb9fed0b3919/dev/bots/test.dart#L715-L716
https://github.com/flutter/flutter/blob/5cc7b6c3ea5fd76111920fec8803712e06fda98c/dev/devicelab/lib/tasks/perf_tests.dart
https://github.com/flutter/flutter/blob/94a41cde83695b1267d0d7fae9d075bfd92af402/dev/devicelab/bin/tasks/image_list_reported_duration.dart
https://github.com/flutter/flutter/blob/94a41cde83695b1267d0d7fae9d075bfd92af402/dev/devicelab/bin/tasks/image_list_jit_reported_duration.dart

However, I will let the team decide if can update `/examples` or not

**Comment by GroovinChip** [comment]
Fair enough @TahaTesser I'll see if I can ensure that my changes don't break the tests. So far I've added macOS support to the hello_world example and tests are passing within that application. I'll see about running more global tests.

**Comment by TahaTesser** [comment]
@GroovinChip 
I am assigning you in the meantime, thanks for this. I hope we can add desktop support for these examples, just have to be sure it doesn't break anything else

**Comment by GroovinChip** [comment]
Absolutely!

**Comment by TahaTesser** [comment]
@GroovinChip 
I just noticed some of these could target the particular [platform ](https://github.com/flutter/flutter/blob/5cc7b6c3ea5fd76111920fec8803712e06fda98c/dev/devicelab/bin/tasks/devtools_profile_start_test.dart#L13)which might not be affected by your change.

If there are such tests, that are only on target Android or iOS, that would be good for your PR(s) examples.  Like I said Team can add more here. I recommend you split PR for each example you modify into separate PRs for easier review and testing

**Comment by GroovinChip** [comment]
Yes, I discovered that as well once I started digging in. I think the hello_world example is the most straightforward, so I'm focusing on that for now. Split PR's is an excellent idea, thank you.

**Comment by GroovinChip** [comment]
@TahaTesser wanted to let you know that  @Abhishek01039 is going to be working with me on this, can we assign him to this issue too?

**Comment by TahaTesser** [comment]
@GroovinChip 
I tried but it replaces you, you can file PR in joint with him, both of you can sign CLA for your commits

**Comment by Abhishek01039** [comment]
Hello @TahaTesser 
me and @GroovinChip can give the MR separately

**Comment by gspencergoog** [comment]
cc @cbracken

**Comment by gspencergoog** [comment]
@GroovinChip Is this something you're continuing to work on?  If so, could you post an update with what else still needs doing here? (Desktop Triage)

**Comment by gspencergoog** [comment]
No worries. Just wanted to make sure you knew you weren't assigned to it anymore (don't want you to work away at something and then find out someone else had done it in the meantime).

**Comment by GroovinChip** [comment]
Thank you!

**Comment by huycozy** [comment]
I think this issue should be split into separate ones for each desktop platform and consider this as an umbrella issue.

**Comment by cbracken** [comment]
Yeah, makes sense now that we've split up the desktop labels. I'll file the issues today and we can probably just close this one altogether.

**Comment by goderbauer** [comment]
(triage): Assigned to cbracken to split this up.

**Comment by darshankawar** [comment]
I will create respective desktop specific issues based on above conversation, as I don't see such issues filed, and will close this in favor of those.
